packages/ros2idl-parser/src/parseRos2idl.ts

```diff
@@ -11,8 +11,15 @@ export function parseRos2idl(messageDefinition: string): MessageDefinition[] {
   // conform to conforming idl and just read it all in a single parse so that we don't have to call parse multiple times
   const idlConformedDef = messageDefinition.replaceAll(ROS2IDL_HEADER, "");
 
-  const results = parseIdl(idlConformedDef);
+  const rawDefinitions = parseIdlToAST(idlConformedDef);
 
+  const idlProcessor = new IDLNodeProcessor(rawDefinitions);
+  idlProcessor.resolveEnumTypes();
```

Comment:
What's the benefit of having to call all these separate functions? What happens if you forget to call these? Do you end up with malformed or incomplete message definitions? If the answer is yes then I think its worth making it easier to do the right thing. Mayhe that's what this `parseRos2Idl` does.

Something else worth noting is whether you really want to make `IDLNodeProcessor` part of the public interface.